File: frontend/src/api/project-member.ts
import api from './index'

export interface ProjectMemberItem {
  id?: number
  projectId: number
  userId: number
  roleInProject: string
  memberLine?: string      // TECHNICAL/ADMINISTRATIVE/QUALITY/CRAFT
  memberPosition?: string  // CHIEF_DESIGNER/CHIEF_COMMANDER/...
  supervisorId?: number    // 上级 member_id
  sortOrder?: number
  duties: string
  status: string
  // 联表查询时可能返回的扩展字段
  userName?: string
  userOrgName?: string
  userTitle?: string
}

export function getProjectMembers(projectId: number) {
  return api.get(`/project-members/project/${projectId}`)
}

export function createProjectMember(data: ProjectMemberItem) {
  return api.post('/project-members', data)
}

export function updateProjectMember(id: number, data: ProjectMemberItem) {
  return api.put(`/project-members/${id}`, data)
}

export function deleteProjectMember(id: number) {
  return api.delete(`/project-members/${id}`)
}

File: backend/src/main/java/com/military/doc/modules/project/entity/ProjectMember.java
package com.military.doc.modules.project.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("project_member")
public class ProjectMember {
    @TableId(type = IdType.AUTO)
    private Long id;

    private Long projectId;
    private Long userId;
    private String roleInProject;
    /** 指挥线: TECHNICAL/ADMINISTRATIVE/QUALITY/CRAFT */
    private String memberLine;
    /** 岗位编码: CHIEF_DESIGNER/CHIEF_COMMANDER/... */
    private String memberPosition;
    /** 上级 member_id，构建指挥链 */
    private Long supervisorId;
    private Integer sortOrder;
    private String duties;
    private String status;

    @TableField(fill = FieldFill.INSERT)
    private Long createdBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Long updatedBy;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableLogic
    private Integer deleted;
}